### PR TITLE
Fix crash after paste and unmount

### LINKED ISSUE #100585: Crash after toolbar paste
If you paste into a text field and then immediately remove the field, the app will crash.

This was reported as a Google internal bug at b/224914489.

<details>

<summary>Error logs</summary>

```
The following _CastError was thrown during a scheduler callback:
Null check operator used on a null value

When the exception was thrown, this was the stack:
#0      EditableTextState.renderEditable (package:flutter/src/widgets/editable_text.dart:2889:67)
#1      EditableTextState.bringIntoView (package:flutter/src/widgets/editable_text.dart:2911:28)
#2      EditableTextState.pasteText.<anonymous closure> (package:flutter/src/widgets/editable_text.dart:1744:11)
#3      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1145:15)
#4      SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1090:9)
#5      SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:996:5)
#9      _invoke (dart:ui/hooks.dart:151:10)
#10     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:308:5)
#11     _drawFrame (dart:ui/hooks.dart:115:31)
(elided 3 frames from dart:async)
```

</details>

I'm reproducing this by using a custom paste button, but this could also happen if you're changing state after the field is filled or something like that.

### How to reproduce

1. Run the app below.  The key is that it must have a text field that is removed on paste.
2. Right click or longpress it to show the text selection menu.
3. Tap paste.

Expected: The clipboard contents are pasted.
Actual: The app crashes.

<details>

<summary>Code</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatefulWidget {
  const MyApp({ Key? key }) : super(key: key);

  @override _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  bool _fieldIsVisible = true;

  void _toggleFieldVisible() {
    setState(() {
      _fieldIsVisible = !_fieldIsVisible;
    });
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: !_fieldIsVisible ? const Placeholder() : TextField(
          maxLines: 5,
          selectionControls: MyMaterialTextSelectionControls(
            onPaste: _toggleFieldVisible,
          ),
        ),
      ),
    );
  }
}

class MyMaterialTextSelectionControls extends MaterialTextSelectionControls {
  MyMaterialTextSelectionControls({
    required this.onPaste,
  });

  final VoidCallback onPaste;

  // Padding between the toolbar and the anchor.
  static const double _kToolbarContentDistanceBelow = 20.0;
  static const double _kToolbarContentDistance = 8.0;

  @override
  Future<void> handlePaste(TextSelectionDelegate delegate) {
    onPaste();
    return super.handlePaste(delegate);
  }

  /// Builder for material-style copy/paste text selection toolbar.
  @override
  Widget buildToolbar(
    BuildContext context,
    Rect globalEditableRegion,
    double textLineHeight,
    Offset position,
    List<TextSelectionPoint> endpoints,
    TextSelectionDelegate delegate,
    ClipboardStatusNotifier? clipboardStatus,
    Offset? lastSecondaryTapDownPosition,
  ) {
    final Offset selectionMidpoint = position;
    final TextSelectionPoint startTextSelectionPoint = endpoints[0];
    final TextSelectionPoint endTextSelectionPoint = endpoints.length > 1
      ? endpoints[1]
      : endpoints[0];
    final Offset anchorAbove = Offset(
      globalEditableRegion.left + selectionMidpoint.dx,
      globalEditableRegion.top + startTextSelectionPoint.point.dy - textLineHeight - _kToolbarContentDistance
    );
    final Offset anchorBelow = Offset(
      globalEditableRegion.left + selectionMidpoint.dx,
      globalEditableRegion.top + endTextSelectionPoint.point.dy + _kToolbarContentDistanceBelow,
    );

    return MyTextSelectionToolbar(
      anchorAbove: anchorAbove,
      anchorBelow: anchorBelow,
      handlePaste: canPaste(delegate) ? () => handlePaste(delegate) : null,
    );
  }
}

class MyTextSelectionToolbar extends StatefulWidget {
  const MyTextSelectionToolbar({
    Key? key,
    required this.anchorAbove,
    required this.anchorBelow,
    this.handlePaste,
  }) : super(key: key);

  final Offset anchorAbove;
  final Offset anchorBelow;
  final VoidCallback? handlePaste;

  @override
  MyTextSelectionToolbarState createState() => MyTextSelectionToolbarState();
}

class MyTextSelectionToolbarState extends State<MyTextSelectionToolbar> {
  @override
  Widget build(BuildContext context) {
    assert(debugCheckHasMaterialLocalizations(context));
    final MaterialLocalizations localizations = MaterialLocalizations.of(context);

    return TextSelectionToolbar(
      anchorAbove: widget.anchorAbove,
      anchorBelow: widget.anchorBelow,
      toolbarBuilder: (BuildContext context, Widget child) {
        return Container(
          color: Colors.pink,
          child: child,
        );
      },
      children: <Widget>[
        TextSelectionToolbarTextButton(
          padding: TextSelectionToolbarTextButton.getPadding(0, 1),
          onPressed: widget.handlePaste,
          child: Text(localizations.pasteButtonLabel),
        ),
      ],
    );
  }
}
```

</details>

### PR DESCRIPTION
**Pull request by justinmc**
Previously, pasting and immediately removing the field caused a crash.  This was caused by https://github.com/flutter/flutter/pull/98604 adding an addPostFrameCallback without checking `mounted`.  This PR simply checks mounted before doing anything in the post frame callback.

Fixes https://github.com/flutter/flutter/issues/100585

### CODE REVIEW

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 1]
```diff
@@ -11731,6 +11731,55 @@ void main() {
...
+    await tester.pump();
+    expect(state.showToolbar(), isTrue);
+    await tester.pumpAndSettle();
+    expect(find.text('Paste'), findsOneWidget);
+
+    await tester.tap(find.text('Paste'));
+    await tester.pumpAndSettle();
+
+    expect(tester.takeException(), null);
+  }, skip: kIsWeb); // [intended]
```
Comment (chunhtai) [inline packages/flutter/test/widgets/editable_text_test.dart:11782]: Can you put some more explanations or a tracking issue?

**Review by chunhtai — APPROVED** [review]
LGTM

**Comment by tgucio** [comment]
@justinmc I guess the same issue is in cutSelection():

```
  void cutSelection(SelectionChangedCause cause) {
    [...]
    if (cause == SelectionChangedCause.toolbar) {
      // Schedule a call to bringIntoView() after renderEditable updates.
      SchedulerBinding.instance.addPostFrameCallback((_) {
        bringIntoView(textEditingValue.selection.extent);
      });
```

**Comment by justinmc** [comment]
@tgucio Ah of course, thanks for the reminder!  I'll put up a PR to fix that too.

**Pull request merged**